presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -302,18 +303,17 @@ public Object visitFieldReference(FieldReference node, Object context)
         @Override
         protected Object visitDereferenceExpression(DereferenceExpression node, Object context)
         {
-            Type returnType = type(node);
-            Optional<Object> maybeEnumValue = tryResolveEnumLiteral(node, returnType);
-            if (maybeEnumValue.isPresent()) {
-                return maybeEnumValue.get();
-            }
-
             Type type = type(node.getBase());
```

Comment:
so in the case of an enum literal `foo.bar.Color.BLUE`, both `foo.bar.Color` and `foo.bar.Color.BLUE` would be of type `ColorEnum` ?

Reply:
Yes, which makes sense right? You did set both nodes' type to the enum.